### PR TITLE
Added Serializable to all classes with a capnp write function

### PR DESCRIPTION
**Pull request by rhyolight**
There is a problem with this PR, tests will fail with this error, but I'm not sure why:

```
============================================== ERRORS ==============================================
_________________________ ERROR collecting tests/unit/nupic/utils_test.py __________________________
tests/unit/nupic/utils_test.py:28: in <module>
    from nupic.utils import MovingAverage
src/nupic/utils.py:32: in <module>
    class MovingAverage(object, Serializable):
/System/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/abc.py:87: in __new__
    cls = super(ABCMeta, mcls).__new__(mcls, name, bases, namespace)
E   TypeError: Error when calling the metaclass bases
E       Cannot create a consistent method resolution
E   order (MRO) for bases object, Serializable
!!!!!!!!!!!!!!!!!!!!!!!!!!!!!! Interrupted: stopping after 1 failures !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
===================================== 1 error in 0.22 seconds ======================================
```

### CODE REVIEW

**Comment by lscheinkman** [comment]
The [MovingAverage](https://github.com/rhyolight/nupic/blob/d94ff7f4f747fbe6d6e994e908cb39fcf51a6880/src/nupic/utils.py#L32) class does not implement the abstract method [getSchema](
https://github.com/rhyolight/nupic/blob/d94ff7f4f747fbe6d6e994e908cb39fcf51a6880/src/nupic/serializable.py#L36-L38)

**Comment by rhyolight** [comment]
@lscheinkman Even after that change, I still get the same error on MovingAverage.

**Comment by lscheinkman** [comment]
remove `object` from the mixin list. `Serializable` already inherits from `object`.
```
class MovingAverage(Serializable):
  """
```

**Comment by rhyolight** [comment]
Thanks @lscheinkman that was it.

File: src/nupic/utils.py [hunk 1]
```diff
@@ -127,3 +128,9 @@ def write(self, proto):
     proto.windowSize = self.windowSize
     proto.slidingWindow = self.slidingWindow
     proto.total = self.total
+
+
+  @classmethod
+  def getSchema(cls):
+    raise NotImplementedError()
```
Comment (lscheinkman) [inline src/nupic/utils.py]: Should return `MovingAverageProto` from this import
```
from nupic.movingaverage_capnp import MovingAverageProto
```

**Comment by rhyolight** [comment]
@lscheinkman ready

**Comment by rhyolight** [comment]
investigating test failure 🤷‍♂️

**Comment by rhyolight** [comment]
Windows built error doesn't make sense. Re-running to see what happens.

**Comment by lscheinkman** [comment]
You need to add capnp conditionals for Windows. Something like this:
```
try:
   import capnp
 except ImportError:
   capnp = None
if capnp:
  from nupic.movingaverage_capnp import MovingAverageProto
```

**Comment by rhyolight** [comment]
@lscheinkman thanks for the help

**Pull request merged**